### PR TITLE
Support Gradle and Android 7.0

### PR DESCRIPTION
**Pull request by vsbogd**
List of changes which were done to support Gradle and Android starting from 7.0:
- Get rid of using command line web3j to generate Java stubs for Solidity contracts
- Move SDK plugin core into separate project `plugin-core`
- Add unit tests for SDK plugin core
- Add Gradle SDK plugin to support Android platform
- Add instruction about using SDK in Gradle
- Add own base64 implementation and tests to support Android 7.0
- Remove Guava usage as it is conflicted with Gradle Android plugin
- Downgrade `web3j` version to `4.2.0-android` to support Android 7.0

Changes are relatively simple and organized by commits. It is better to review this PR commit by commit.

### CODE REVIEW

File: plugin-core/src/main/java/io/singularitynet/sdk/plugin/ServiceApiGetter.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                if (entry.isDirectory()) {
+                    if (!f.isDirectory() && !f.mkdirs()) {
+                        throw new IOException("failed to create directory " + f);
+                    }
+                } else {
+                    File parent = f.getParentFile();
+                    if (!parent.isDirectory() && !parent.mkdirs()) {
+                        throw new IOException("failed to create directory " + parent);
+                    }
+                    try (OutputStream o = Files.newOutputStream(f.toPath())) {
```
Comment (noskill) [inline plugin-core/src/main/java/io/singularitynet/sdk/plugin/ServiceApiGetter.java]: Does a **try** without **catch** does anything?

Comment (vsbogd) [reply plugin-core/src/main/java/io/singularitynet/sdk/plugin/ServiceApiGetter.java]: Yes, it is `try with` block, it closes `OutputStream o` after leaving the block. So it is equal to 
```java
OutputStream o = Files.newOutputStream(f.toPath());
try {
} finally {
  o.close();
}
```

**Comment by noskill** [comment]
looks good to me

**Comment by vsbogd** [comment]
Thanks @noskill !

**Pull request merged**